python/cuml/common/array.py

```diff
@@ -137,26 +139,45 @@ def __init__(self,
 
         ary_interface = False
 
-        # Base class (Buffer) constructor call
         size, shape = _get_size_from_shape(shape, dtype)
 
         if not memview_construction and not detailed_construction:
-            # Convert to cupy array and manually specify the ptr, size and
-            # owner. This is to avoid the restriction on Buffer that requires
-            # all data be u8
             cupy_data = cp.asarray(data)
-            flattened_data = cupy_data.data.ptr
 
             # Size for Buffer is not the same as for cupy. Use nbytes
             size = cupy_data.nbytes
             owner = cupy_data if cupy_data.flags.owndata else data
-        else:
-            flattened_data = data
+            data = cupy_data
 
         self._index = index
-        super().__init__(data=flattened_data,
-                         owner=owner,
-                         size=size)
+
+        if hasattr(data, "__array_interface__"):
+            ary_interface = data.__array_interface__
+        elif hasattr(data, "__cuda_array_interface__"):
+            ary_interface = data.__cuda_array_interface__
+
+        # We no longer call the super init due to the refactor
+        if isinstance(data, int):
```

Comment:
Is int the only numeric type that we'd want to handle here?

Reply:
In a future refactor we might want to accept other forms of specifying pointers for construction, but we don't do that currently